Transient `test_get_cached_manifest` unit test failure

… first observed in a GitLab test job: https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/25858#L3373

Then happened again for a GitHub actions unit test run: https://github.com/DataBiosphere/azul/actions/runs/6511198193/job/17686420982#step:4:8503

```
 ======================================================================
FAIL: test_get_cached_manifest (service.test_manifest.TestManifestCache)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/azul/azul/.venv/lib/python3.9/site-packages/moto/core/models.py", line 128, in wrapper
    result = func(*args, **kwargs)
  File "/home/runner/work/azul/azul/.venv/lib/python3.9/site-packages/moto/core/models.py", line 128, in wrapper
    result = func(*args, **kwargs)
  File "/home/runner/work/azul/azul/test/service/test_manifest.py", line 191, in wrapper
    return test(self, *args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.9.17/x64/lib/python3.9/unittest/mock.py", line 1336, in patched
    return func(*newargs, **newkeywargs)
  File "/home/runner/work/azul/azul/test/service/test_manifest.py", line 1422, in test_get_cached_manifest
    self.assertEqual(cached_manifest_1, cached_manifest_2)
AssertionError: Manif[377 chars]ture=op61Erg8lQaZjKLM0EzecCQI9EE%3D&x-amz-secu[393 chars]lrc') != Manif[377 chars]ture=D%2FM%2Fk%2FC0DDBuyDkHg0ZK8vEftXw%3D&x-am[399 chars]lrc')

----------------------------------------------------------------------
Ran 402 tests in 1355.536s

FAILED (failures=1, skipped=4)
make: *** [Makefile:218: test] Error 1
Error: Process completed with exit code 2.
```
<!--
Add your description here. For suspected bugs, the steps required to reproduce
the problem, as well as the expected and actual outcome should be included in
the description. Solutions should be proposed in a  separate comment, not the
issue description. Please do not modify or delete the checklist below. The
checklist should remain at the end of the issue description.
-->

- [ ] Security design review completed; the Resolution of this issue does **not** …
  - [ ] … affect authentication; for example:
    - OAuth 2.0 with the application (API or Swagger UI)
    - Authentication of developers with Google Cloud APIs
    - Authentication of developers with AWS APIs
    - Authentication with a GitLab instance in the system
    - Password and 2FA authentication with GitHub
    - API access token authentication with GitHub
    - Authentication with 
  - [ ] … affect the permissions of internal users like access to
    - Cloud resources on AWS and GCP
    - GitLab repositories, projects and groups, administration
    - an EC2 instance via SSH
    - GitHub issues, pull requests, commits, commit statuses, wikis, repositories, organizations
  - [ ] … affect the permissions of external users like access to
    - TDR snapshots
  - [ ] … affect permissions of service or bot accounts
    - Cloud resources on AWS and GCP
  - [ ] … affect audit logging in the system, like
    - adding, removing or changing a log message that represents an auditable event
    - changing the routing of log messages through the system
  - [ ] … affect monitoring of the system
  - [ ] … introduce a new software dependency like
    - Python packages on PYPI
    - Command-line utilities
    - Docker images
    - Terraform providers
  - [ ] … add an interface that exposes sensitive or confidential data at the security boundary
  - [ ] … affect the encryption of data at rest
  - [ ] … require persistence of sensitive or confidential data that might require encryption at rest
  - [ ] … require unencrypted transmission of data within the security boundary
  - [ ] … affect the network security layer; for example by 
    - modifying, adding or removing firewall rules
    - modifying, adding or removing security groups
    - changing or adding a port a service, proxy or load balancer listens on
- [ ] Documentation on any unchecked boxes is provided in comments below

--- a/test/service/test_manifest.py
+++ b/test/service/test_manifest.py
@@ -1414,11 +1414,18 @@ class TestManifestCache(ManifestTestCase):
                                                               catalog=manifest_key.catalog,
                                                               filters=filters)
         self.assertTrue(cached_manifest_1.was_cached)
-        # The was_cached property should be the only difference
-        manifest = attrs.evolve(manifest, was_cached=True)
+        # The `was_cached` and `location` properties should be the only
+        # differences. The `location` is a signed S3 URL that depends on
+        # the current time. If both manifest where created in different
+        # seconds, the signed URL is going to have a different expiration.
+        manifest = attrs.evolve(manifest,
+                                was_cached=True,
+                                location=cached_manifest_1.location)
         self.assertEqual(manifest, cached_manifest_1)
 
         cached_manifest_2 = self._service.get_cached_manifest_with_key(manifest_key)
+        cached_manifest_1 = attrs.evolve(cached_manifest_1,
+                                         location=cached_manifest_2.location)
         self.assertEqual(cached_manifest_1, cached_manifest_2)
 
         # Ensure manifest is considered stale

